### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by srudaa**
### Added

- Prometheus metrics server exposing Lora metadata 

### Changed

- Async Handlers

### Fixed

- Logging methods

### CODE REVIEW

File: Source/Model/RxMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /// <summary>
+    /// Metadata for each antenna of each gateway that received this message
+    /// </summary>
+
+    [JsonObject(NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
+    public class RxMetadata
+    {
+        /// <summary>
+        /// Define application id
```
Comment (jonasmwahl) [inline Source/Model/RxMetadata.cs:17]: Define 'Gateway id', not 'application id'

File: Source/MetricsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        /// <summary>
+        /// Prometheus gauge showing the Lora Signal-to-Noise Ratio per gateway
+        /// <see href="https://lora.readthedocs.io/en/latest/#snr"></see>
+        /// </summary>
+        private static readonly Gauge Snr = Metrics.CreateGauge("raaedge_connector_lora_snr", "Lora signal to noise ratio", new GaugeConfiguration { LabelNames = new[] { "iothub", "edge_device", "application_id", "dev_eui", "device_id", "gateway_id", "gateway_eui" } });
+        
+        /// <summary>
+        /// Prometheus gauge showing the Lora frame counter
+        /// </summary>
+        private static readonly Gauge FCnt = Metrics.CreateGauge("raaedge_connector_lora_fcnt", "Lora message frame counter", new GaugeConfiguration { LabelNames = new[] { "iothub", "edge_device", "application_id", "dev_eui", "device_id"} });
```
Comment (jonasmwahl) [inline Source/MetricsHandler.cs:49]: Could this be a [Counter](https://prometheus.io/docs/concepts/metric_types/#counter)

**Review by jonasmwahl — CHANGES_REQUESTED** [review]
One comment on a prometheus metrics

File: Source/MetricsHandler.cs (hunk 2 above)
Comment (srudaa) [reply Source/MetricsHandler.cs:49]: No. It needs to be gauge since the frame counter is just another value that can increase or decrease.

**Review by jonasmwahl — APPROVED** [review]

**Pull request merged**